turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -689,7 +735,11 @@ impl<B: Backend + 'static> TurboTasks<B> {
         let future = async move {
             let mut schedule_again = true;
             while schedule_again {
-                let global_task_state = Arc::new(RwLock::new(CurrentGlobalTaskState::new(task_id)));
+                let backend_state = this.backend.new_task_state(task_id);
+                let global_task_state = Arc::new(RwLock::new(CurrentGlobalTaskState::new(
+                    task_id,
+                    Box::new(backend_state),
```

Comment:
This PR adds an extra allocation for every task...
Could this use generics and `B::TaskState` to avoid the Box?

Reply:
Ok sounds good. The new backend will get rid of `dependencies_to_track` anyway, so we might be able to remove the whole TaskState eventually.